src/graphnet/data/datamodule.py

```diff
@@ -72,6 +78,78 @@ def __init__(
 
         self.setup("fit")
 
+    def _set_dataloader_kwargs(
+        self,
+        train_dl_args: Dict[str, Any],
+        val_dl_args: Union[Dict[str, Any], None],
+        test_dl_args: Union[Dict[str, Any], None],
+    ) -> None:
+        """Copy train dataloader args to other dataloaders if not given.
+
+        Also checks that ParquetDataset dataloaders have multiprocessing
+        context set to "spawn" as this is strictly required.
+
+        See: https://docs.pola.rs/user-guide/misc/multiprocessing/
+        """
+        if val_dl_args is None:
+            self.info(
+                "No `val_dataloader_kwargs` given. This arg has "
+                "been set to `train_dataloader_kwargs` automatically."
+            )
+            val_dl_args = deepcopy(train_dl_args)
+        if (test_dl_args is None) & (self._test_selection is not None):
+            test_dl_args = deepcopy(train_dl_args)
+            self.info(
+                "No `test_dataloader_kwargs` given. This arg has "
+                "been set to `train_dataloader_kwargs` automatically."
+            )
+
+        if self._dataset == ParquetDataset:
+            train_dl_args = self._add_context(train_dl_args, "training")
+            val_dl_args = self._add_context(val_dl_args, "validation")
+            if self._test_selection is not None:
+                assert test_dl_args is not None
+                test_dl_args = self._add_context(test_dl_args, "test")
+
+        self._train_dataloader_kwargs = train_dl_args
+        self._validation_dataloader_kwargs = val_dl_args
+        self._test_dataloader_kwargs = test_dl_args or {}
+
+    def _add_context(
+        self, dataloader_args: Dict[str, Any], dataloader_type: str
+    ) -> Dict[str, Any]:
+        """Handle assignment of `multiprocessing_context` arg to loaders.
+
+        Datasets relying on threaded libraries often require the
+        multiprocessing context to be set to "spawn". This method will check
+        the arguments for this entry and throw an error if the field is already
+        assigned to a wrong value. If the value is not specified, it is added
+        automatically with a log entry.
+        """
+        arg = "multiprocessing_context"
+        if arg in dataloader_args:
```

Comment:
Maybe it is better to define the case where `num_workers=0`, then no multiprocessing is needed, and this arg will trigger the error:

```
    dataloader = DataLoader(
  File "~/python3.10/site-packages/torch/utils/data/dataloader.py", line 262, in __init__
    self.multiprocessing_context = multiprocessing_context
  File "~/python3.10/site-packages/torch/utils/data/dataloader.py", line 421, in __setattr__
    super().__setattr__(attr, val)
  File "~/python3.10/site-packages/torch/utils/data/dataloader.py", line 410, in multiprocessing_context
    raise ValueError('multiprocessing_context can only be used with '
ValueError: multiprocessing_context can only be used with multi-process loading (num_workers > 0), but got num_workers=0
```